### PR TITLE
Deprecate old HTTP wrapping methods and add instructions for installing dd-trace-java.

### PR DESCRIPTION
**Pull request by agocs**
### What does this PR do?

Adds installation instructions for the Java tracer. Marks some functions deprecated that the tracer does better.

### Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Checklist

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,191 +38,121 @@ Once [installed](#installation), you should be able to submit custom metrics fro
...
 
-- java.net.HttpUrlConnection
-- Apache HTTP Client
-- OKHttp3
+### Cold start considerations
 
-Don't see your favorite client? Open an issue and request it. Datadog is adding to 
-this library all the time.
+The Java Tracer adds a nontrivial cold start penalty. 
+Expect roughly 6 seconds per cold start if your lambda function is configured with 3008MB of memory.
```
Comment (nhinsch) [inline README.md:50]: Nit: lambda -> Lambda

File: README.md [hunk 2]
```diff
@@ -38,191 +38,121 @@ Once [installed](#installation), you should be able to submit custom metrics fro
...
+Lambda runtime CPU scales with the amount of memory allocated, so allocating more memory may  help alleviate cold start issues.
+Also consider using provisioned concurrency to keep your lambda function warm.
 
-### HttpUrlConnection examples
+### Compatable Java runtimes
```
Comment (nhinsch) [inline README.md:54]: Compatable -> Compatible

File: README.md [hunk 3]
```diff
@@ -38,191 +38,121 @@ Once [installed](#installation), you should be able to submit custom metrics fro
...
+If you are using a different event with trace context, you may choose to create a class that implements `Headerable` and supply that as the event instead.
 
-#### Plain text logs
+## Downstream Requests
 
-If you are using plain text logs, then you must create a new [Parser](https://docs.datadoghq.com/logs/processing/parsing/?tab=matcher)
-by cloning the existing Lambda Pipeline. The new parser can extract the trace context from the correct position in the logs. 
-Use the helper `_trace_context` to extract the trace context. For example, if your log line looked like:
+The dd-trace-java tracer will automatically add trace context to outgoing requests for a number of popular services. 
+The list of instrument services can be found here: https://docs.datadoghq.com/tracing/setup_overview/compatibility_requirements/java/ .
```
Comment (nhinsch) [inline README.md:122]: instrument -> instrumented

File: README.md [hunk 4]
```diff
@@ -38,191 +38,121 @@ Once [installed](#installation), you should be able to submit custom metrics fro
...
 
-#### Other logging solutions
+The following grok parser parses Java logs formatted using the pattern in the previous section.
 
-If you are using a different logging solution, the trace ID can be accessed using the method
-`DDLambda.getTraceContextString()`. That returns your trace ID as a string that can be added
-to any log message.
+```
+java_tracer %{date("yyyy-MM-dd HH:mm:ss"):timestamp}\s\<%{uuid:lambda.request_id}\>\s%{word:level}\s+%{data:call_site}%{_trace_context}%{data:message}
+```
```
Comment (nhinsch) [inline README.md:155]: Is this parser the same one we provide out of the box with the default Lambda log pipeline? If that is the case, shouldn't this just work without the customer having to know about the parser? If that is **not** the case, is there a reason we can't format the logs in the format already understood by the default parser?

**Review by nhinsch — APPROVED** [review]
Looks good to me, just did a proofreading pass, also one question about the logs parser.

File: README.md (hunk 4 above)
Comment (agocs) [reply README.md:155]: We don't have the power (as far as I know) to change how the customer formats their logs. We can just suggest formats. I think the smart thing to do is to bias towards action, get this out there, and see how customers use it.

**Pull request merged**